File: ros1bookcode-master-rw/chapt2/chapt2_ws/src/demo_cpp_pkg/src/person_node.cpp
#include <string>
#include <ros/ros.h>

class PersonNode
{
private:
  ros::NodeHandle nh_;
  std::string name_;
  int age_;

public:
  PersonNode(const std::string &name, int age) : name_(name), age_(age) {}

  void eat(const std::string &food_name)
  {
    ROS_INFO("我是%s，今年%d岁，我现在正在吃%s", name_.c_str(), age_, food_name.c_str());
  }
};

int main(int argc, char **argv)
{
  ros::init(argc, argv, "person_node");
  PersonNode node("法外狂徒张三", 18);
  node.eat("鱼香ROS");
  ros::spin();
  return 0;
}

File: ros1bookcode-master-rw/chapt2/chapt2_ws/src/demo_python_pkg/demo_python_pkg/person_node.py
#!/usr/bin/env python3
import rospy

class PersonNode:
    def __init__(self, node_name: str, name: str, age: int) -> None:
        rospy.init_node(node_name, anonymous=False)
        self.age = int(age)
        self.name = name

    def eat(self, food_name: str):
        rospy.loginfo(f'我叫{self.name},今年{self.age}岁，我现在正在吃{food_name}')

def main():
    node = PersonNode('person_node', '法外狂徒张三', 18)
    node.eat('鱼香肉丝')
    rospy.spin()

if __name__ == '__main__':
    main()
